Stepic/Services/Deep Links/BranchService.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  BranchService.swift
+//  Stepic
+//
+//  Created by Ostrenkiy on 12/11/2018.
+//  Copyright © 2018 Alex Karpov. All rights reserved.
+//
+
+import Foundation
+import Branch
+
+class BranchService {
+    private var deepLinkRoutingService: DeepLinkRoutingService
+
+    init(deepLinkRoutingService: DeepLinkRoutingService) {
+        self.deepLinkRoutingService = deepLinkRoutingService
+    }
+
+    func setup(launchOptions: [UIApplicationLaunchOptionsKey: Any]?) {
+        Branch.getInstance().initSession(launchOptions: launchOptions) { params, _ in
+            guard let data = params as? [String: AnyObject] else {
+                return
+            }
+
+            DispatchQueue.main.async { [weak self] in
+                self?.deepLinkRoutingService.route(DeepLinkRoute(data: data))
+            }
+        }
+    }
+
+    func openURL(userActivity: NSUserActivity) {
+        Branch.getInstance()?.continue(userActivity)
+    }
+
+    func openURL(_ app: UIApplication, open url: URL, options: [UIApplication.OpenURLOptionsKey : Any]) {
+        Branch.getInstance().application(app, open: url, options: options)
+    }
+
+    func isBranch(url: URL) -> Bool {
```

Comment:
Может быть лучше более понятное название
```suggestion
    func isBranchURL(_ url: URL) -> Bool {
```
или
```suggestion
    func canOpenWithBranch(url: URL) -> Bool {
```